tensorflow/contrib/eager/python/examples/generative_examples/dcgan.ipynb

```diff
@@ -1,694 +1,928 @@
+    "id": "2MbKJY38Puy9"
+   },
+   "source": [
+    "## What are GANs?\n",
+    "GANs, or [Generative Adversarial Networks](https://arxiv.org/abs/1406.2661), are a framework for estimating generative models. Two models are trained simultaneously by an adversarial process: a Generator, which is responsible for generating data (say, images), and a Discriminator, which is responsible for estimating the probability that an image was drawn from the training data (the image is real), or was produced by the Generator (the image is fake). During training, the Generator becomes progressively better at generating images, until the Discriminator is no longer able to distinguish real images from fake. \n",
+    "\n",
+    "![alt text](https://github.com/margaretmz/tensorflow/blob/margaret-dcgan/tensorflow/contrib/eager/python/examples/generative_examples/gans_diagram.png?raw=1)\n",
+    "\n",
+    "We will demonstrate this process end-to-end on MNIST. Below is an animation that shows a series of images produced by the Generator as it was trained for 150 epochs. Overtime, the generated images become increasingly difficult to distinguish from the training set.\n",
+    "\n",
+    "To learn more about GANs, we recommend MIT's [Intro to Deep Learning](http://introtodeeplearning.com/) course, which includes a lecture on Deep Generative Models ([video](https://youtu.be/JVb54xhEw6Y) | [slides](http://introtodeeplearning.com/materials/2018_6S191_Lecture4.pdf)). Now, let's head to the code!\n",
+    "\n",
+    "![sample output](https://tensorflow.org/images/gan/dcgan.gif)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
     "colab": {
-      "collapsed_sections": [],
-      "name": "dcgan.ipynb",
-      "private_outputs": true,
-      "provenance": [
-        {
-          "file_id": "1eb0NOTQapkYs3X0v-zL1x5_LFKgDISnp",
-          "timestamp": 1527173385672
-        }
-      ],
-      "toc_visible": true,
-      "version": "0.3.2"
-    },
-    "kernelspec": {
-      "display_name": "Python 3",
-      "language": "python",
-      "name": "python3"
-    }
-  },
-  "nbformat": 4,
-  "nbformat_minor": 0
+     "base_uri": "https://localhost:8080/",
+     "height": 221
+    },
+    "colab_type": "code",
+    "id": "u_2z-B3piVsw",
+    "outputId": "684f2b6e-7756-448e-da2a-74bcb08d8686"
+   },
+   "outputs": [],
+   "source": [
+    "# Install imgeio in order to generate an animated gif showing the image generating process\n",
+    "!pip install imageio"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "colab_type": "text",
+    "id": "e1_Y75QXJS6h"
+   },
+   "source": [
+    "### Import TensorFlow and enable eager execution"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "colab": {},
+    "colab_type": "code",
+    "id": "YfIk2es3hJEd"
+   },
+   "outputs": [],
+   "source": [
+    "from __future__ import absolute_import, division, print_function\n",
```

Comment:
Can we ditch the future imports? I think they're sad.

Reply:
Removed the future imports.